test/OpenTelemetry.Instrumentation.SqlClient.Tests/SqlClientTestCase.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections;
+
+namespace OpenTelemetry.Instrumentation.SqlClient.Tests;
+
+public class SqlClientTestCase : IEnumerable<object[]>
+{
+    public string ConnectionString { get; set; } = string.Empty;
+
+    public string ExpectedActivityName { get; set; } = string.Empty;
+
+    public string? ExpectedServerAddress { get; set; }
+
+    public int? ExpectedPort { get; set; }
+
+    public string? ExpectedDbNamespace { get; set; }
+
+    public string? ExpectedInstanceName { get; set; }
+
+    private static SqlClientTestCase[] TestCases =>
+    [
+        new SqlClientTestCase
+            {
+                ConnectionString = @"Data Source=SomeServer",
```

Comment:
I will be expanding these test cases, but wanted to model in this PR what I plan to do.

I find the `InlineData` attributes onerous when they include a lot of parameters. It's more code, but I prefer creating a class to represent complex test cases. I find it easier to read and understand the cases covered.